### PR TITLE
fix: Use correct docker image tag after publishing (DEV-614)

### PR DESCRIPTION
**Pull request by irinaschubert**
Resolves DEV-614

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -86,6 +86,11 @@ jobs:
         uses: coursier/cache-action@v6
       - name: generate api-client-test-data
         run: make client-test-data
+      - name: Upload the test output
```
Comment (mpro7) [inline .github/workflows/main.yml:89]:
```suggestion
      - name: Upload client-test-data
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -55,9 +55,8 @@ jobs:
 
 # ------------------------------------------
 # ------------------------------------------
-# Test client-test-data 
   client-test-data-tests:
-    name: Test client-test-data
+    name: Run client-test-data
```
Comment (BalduinLandolt) [inline .github/workflows/main.yml:59]: all other names are now lower case

**Review by BalduinLandolt — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 2 above)
Comment (irinaschubert) [reply .github/workflows/main.yml:59]: Well, if I understood the actual "pattern" (if there was one...) correctly, the name of the action starts with an upper case (it's like the title), the names of the steps with lower case.

Comment (BalduinLandolt) [reply .github/workflows/main.yml:59]: I didn't spot that pattern, but now that you mention it, it makes sense (to some degree)

**Review by subotic — APPROVED** [review]

**Pull request merged**